File: composeApp/src/androidMain/kotlin/com/prekogdevs/fintrack/feature/dashboard/DashboardScreen.kt
package com.prekogdevs.fintrack.feature.dashboard

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material3.FloatingActionButton
import androidx.compose.material3.Icon
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.prekogdevs.fintrack.core.theme.AppColors
import com.prekogdevs.fintrack.feature.dashboard.components.AddTransactionDialog
import com.prekogdevs.fintrack.feature.dashboard.components.BalanceCard
import com.prekogdevs.fintrack.feature.dashboard.components.RecentTransactionsHeader
import com.prekogdevs.fintrack.feature.dashboard.components.SpendingByCategoryCard
import com.prekogdevs.fintrack.feature.dashboard.components.TopBar
import com.prekogdevs.fintrack.feature.dashboard.components.TransactionItem

@Composable
fun DashboardScreen(
    viewModel: AndroidDashboardViewModel = viewModel()
) {
    val state by viewModel.uiState.collectAsState()
    val categoryIconMap = state.availableCategories.associateBy({ it.name }, { it.icon })

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(AppColors.BgGray)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(bottom = 100.dp)
        ) {
            TopBar()
            BalanceCard(state.totalBalance, state.income, state.expenses)
            SpendingByCategoryCard(state.categories)
            RecentTransactionsHeader()

            state.transactions.forEach { transaction ->
                TransactionItem(
                    transaction = transaction,
                    icon = categoryIconMap[transaction.category] ?: ""
                )
            }
        }

        Column(
            modifier = Modifier
                .align(Alignment.BottomEnd)
                .padding(24.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp),
            horizontalAlignment = Alignment.End
        ) {
            FloatingActionButton(
                onClick = { viewModel.onAddClicked() },
                containerColor = AppColors.BluePrimary,
                contentColor = AppColors.BgGray
            ) {
                Icon(Icons.Default.Add, contentDescription = "Add Transaction")
            }
        }

        if (state.showAddDialog) {
            AddTransactionDialog(
                categories = state.availableCategories,
                onDismiss = { viewModel.onDialogDismissed() },
                onAddTransaction = viewModel::onAddTransaction
            )
        }
    }
}

File: iosApp/iosApp/Feature/Dashboard/DashboardScreen.swift
import SwiftUI
import Shared

struct DashboardScreen: View {
    @StateObject private var viewModel = DashboardViewModel()

    var body: some View {
        let categoryIconMap = Dictionary(
            uniqueKeysWithValues: viewModel.availableCategories.map { ($0.name, $0.icon) }
        )

        ZStack(alignment: .bottomTrailing) {
            ScrollView {
                VStack(spacing: 16) {
                    TopBar()
                    BalanceCard(
                        totalBalance: viewModel.totalBalance,
                        income: viewModel.income,
                        expenses: viewModel.expenses
                    )
                    SpendingByCategoryCard(categories: viewModel.categories)
                    RecentTransactionsHeader()

                    ForEach(viewModel.transactions) { transaction in
                        TransactionItem(
                            transaction: transaction,
                            icon: categoryIconMap[transaction.category] ?? ""
                        )
                    }
                }
                .padding(.bottom, 100)
            }
            .background(AppColors.bgGray)

            VStack(spacing: 12) {

                // Add Transaction FAB
                Button(action: { viewModel.onAddClicked() }) {
                    Image(systemName: "plus")
                        .font(.system(size: 24, weight: .bold))
                        .foregroundColor(AppColors.bgGray)
                        .frame(width: 56, height: 56)
                        .background(AppColors.bluePrimary)
                        .clipShape(Circle())
                        .shadow(radius: 4)
                }
            }
            .padding(24)
        }
        .ignoresSafeArea(edges: .bottom)
        .sheet(isPresented: $viewModel.showAddDialog) {
            AddTransactionDialog(
                categories: viewModel.availableCategories,
                onDismiss: { viewModel.onDialogDismissed() },
                onAddTransaction: viewModel.onAddTransaction
            )
            .presentationDetents([.large])
            .presentationDragIndicator(.visible)
        }
    }
}
